coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -83,6 +83,8 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
       materializeSubquery(logicalPlan, qContext)
     } else logicalPlan match {
       case mqp: MetadataQueryPlan               => materializeMetadataQueryPlan(mqp, qContext).plans.head
+      // TODO(a_theimer): awk
+      case lp: TsCardinalities                 => materializeTsCardinalities(lp, qContext).plans.head
       case _                                    => walkLogicalPlanTree(logicalPlan, qContext).plans.head
```

Comment:
The `TsCardinalities` / `MetadataQueryPlan` separation is starting to get a little awkward, especially given the code for `materializeTsCardinalities`. Maybe in a separate PR these should be merged again?